addons/l10n_it_edi/models/account_move.py

```diff
@@ -0,0 +1,458 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import logging
+import re
+
+from odoo import api, fields, models, _
+from odoo.tools import float_compare, float_repr
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_it_edi_state = fields.Selection([
+        ('being_sent', 'Being Sent To SdI'),
+        ('needs_signing', 'Needs signing'),
+        ('processing', 'SdI Processing'),
+        ('rejected', 'SdI Rejected'),
+        ('forwarded', 'SdI Accepted, Forwarded to Partner'),
+        ('forward_failed', 'SdI Accepted, Forward to Partner Failed'),
```

Comment:
Hummmm... no wait... there's a big hole right here! 😭 
`notificaMancataConsegna` should go to `forward_failed` and the one that should go to `forward_failed_attempt` is not even handled by the code! In the image, the message that should bring to `forward_failed_attempt`, has no "State" field but it does have a `SendingAttempts` field which is `TentativiInvio`
![image](https://github.com/odoo/odoo/assets/1665365/b1ee63df-03ac-48e6-a0ae-3b6f63cb7189)

Reply:
Actually, this will need more code in the IAP server to handle the case. This state update is different from other we receive, it's done through the `RicezioneInvoice` API instead of the `TrasmissioneFatture` (even if it's about a transmission) and on the IAP there's no code about this. It's important, because the first forward failed attempt means actually that the SdI accepted the invoice and it's valid legally!!

Reply:
I've added a IAP branch